src/hotspot/share/opto/memnode.cpp

```diff
@@ -1658,35 +1674,41 @@ Node* LoadNode::split_through_phi(PhaseGVN* phase, bool ignore_missing_instance_
 
   // Select Region to split through.
   Node* region;
+  bool dead_code, can_not_split = false;
```

Comment:
```suggestion
  bool dead_code     = false;
  bool can_not_split = false;
```
With your version, would actually both variables be set to `false`, or could one be uninitialized? I just don't know enough about C++ here - but this makes me nervous ;)